src/toil/lib/conversions.py

```diff
@@ -84,3 +88,23 @@ def bytes2human(n: SupportsInt) -> str:
     unit = units[power_level if power_level < len(units) else -1]
     value = convert_units(n, "b", unit)
     return f'{value:.1f} {unit}'
+
+#General Conversions
+
+def hms_duration_to_seconds(hms: str) -> float: 
+    """
+    Parses a given time string in hours:minutes:seconds, 
+    returns an equivalent total seconds value
+    """
+    vals_to_convert = hms.split(':')
+    seconds = 0.0
+    
+    for val in vals_to_convert:
+        if(float(val) < 0): 
+            raise ValueError("Invalid Time")
```

Comment:
Why did you choose to check for negative components but not e.g. the wrong number of colons?

Reply:
Mainly due to the use case, in torque.py we're provided with time in the format of "hh:mm:ss", I think that was probably more because I was approaching the specific case seen in the issue, In retrospect, given that isn't written specifically for this case a error check should be implmented for that, I'll add that in the next commit